source/byte_buf.c

```diff
@@ -750,7 +750,9 @@ int aws_byte_buf_reserve(struct aws_byte_buf *buffer, size_t requested_capacity)
         AWS_POSTCONDITION(aws_byte_buf_is_valid(buffer));
         return AWS_OP_SUCCESS;
     }
-
+    if (!buffer->buffer && !buffer->capacity && requested_capacity > buffer->capacity) {
+        return aws_byte_buf_init(buffer, buffer->allocator, requested_capacity);
```

Comment:
Probably needs an error check and AWS_POSTCONDITION here

Reply:
Why isn't the post condition in aws_byte_buf_init sufficient?

Reply:
Not knowing much about the proof system, I don't know if it is or isn't.